automates/text_reading/src/test/scala/org/clulab/aske/automates/text/TestParameterSetting.scala

```diff
@@ -206,12 +203,155 @@ class TestParameterSetting  extends ExtractionTest {
+    )
+    val mentions = extractMentions(u1a)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  val u2a = "It shows that the pasture is not harvested before 1/07/1995 , the harvest frequency is once every 21 days and the harvest residual is 1250 kg / ha ."
+  failingTest should s"extract the parameter setting(s) from u2a: ${u2a}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "harvest residual" -> Seq("1250") //todo: add this as unit test, too
+    )
+    val mentions = extractMentions(u2a)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  val u3a = "85 % of grazed / harvested nitrogen returns to soil , of which 60 % is in urine and returned to soil profile to a depth of 300 mm ."
+  failingTest should s"extract the parameter setting(s) from u3a: ${u3a}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "soil profile to a depth" -> Seq("300") //fixme: should actually be soil profile depth but this is good enough
+    )
+    val mentions = extractMentions(u3a)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  val u4a = "where sowing depth was known and it is set to 1.5 o Cd per mm ."
+  passingTest should s"extract the parameter setting(s) from u4a: ${u4a}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "sowing depth" -> Seq("1.5")
+    )
+    val mentions = extractMentions(u4a)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  val u4atoy = " where St is sowing depth  and it is set to 1.5 o Cd per mm ."
+  passingTest should s"extract the parameter setting(s) from u4atoy: ${u4atoy}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "St" -> Seq("1.5")
+    )
+    val mentions = extractMentions(u4atoy)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  // todo: add a units test
+  val u5a = "For the purposes of model parameterisation the value of shoot_lag has been assumed to be around 40 o Cd."
+  passingTest should s"extract the parameter setting(s) from u5a: ${u5a}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "value of shoot_lag" -> Seq("40") // fixme: ideally, should just be shoot_lag, but this tests the var-assigned rule
+    )
+    val mentions = extractMentions(u5a)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  //todo: add units test
+  val u6a = "This means that at a sowing depth of 4 cm emergence occurs..."
+  failingTest should s"extract the parameter setting(s) from u6a: ${u6a}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "sowing depth" -> Seq("4")
+    )
+    val mentions = extractMentions(u6a)
+    testParameterSettingEvent(mentions, desired)
+  }
+
+  val u7a = "Phenology APSIM-Barley uses 11 crop stages and ten phases ( time between stages ) ."
+  failingTest should s"extract the parameter setting(s) from u7a: ${u7a}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "crop stages" -> Seq("11"),
+      "phases" -> Seq("ten") // when have started handling word param settings, switch to `10`
```

Comment:
Could you explain what you meant by the above note?

Reply:
like "ten" is not a numerically expressed value. would be nice if we could switch it (and also words like positive/negative) to actual numbers (like ten -> 10, positive is >0, etc)